### DISCUSSION TITLE
Help on using ORY in my app

### DISCUSSION THREAD

**Discussion by NurdinDev** [ory/kratos#1550] [Q&A] [answered]
I'm new to the ORY ecosystem and user Authentication in general, so, please excuse me for my silly question

My app looks like this:
- GraphQL API use Postgress and Prisma ORM
   - I need to secure some endpoint and check if the request has made from the right person or not
   
- Admin dashboard use the GraphQL
   - it should be secured just for specific roles (super admin - admin)
   - and super admin able to manage all users
   
- Client app use the Graphql
   - the client can use the app anonymously 
   - and he can log in and register in my app

I need some help and advice to integrate ORY into my app

Thanks in advance.

**Comment by Benehiko** [comment, marked as answer]
Hi @NurdinDev 

Sorry for the late reply! 

Your question is quite vague in terms of what you have tried to implement or use in the Ory ecosystem. 
I will try give you some pointers which might help you understand the Ory ecosystem better and how it can be implemented in your system. 

Firstly, since you have a login/registration for users you need authentication which [Ory Kratos](https://www.ory.sh/kratos/) solves.
Secondly, you need authorization which [Ory Keto](https://www.ory.sh/keto/) solves. 

In your system you will be calling Ory Kratos (via a middleware) to verify if the user is logged in (and retrieve the logged in users' id) for certain resources which only a registered user can access or that specific users' data. On you Admin dashboard you will use Ory Keto to verify their permissions (e.g. is user A super admin?).

I hope this helps, please let me know if you need any clarification.

**Reply by vinckr** [reply]
Hey @NurdinDev ,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
